test/services/apimap-sender.test.js

```diff
@@ -0,0 +1,75 @@
+const ApplicationContext = require('../../src/context/application-context');
+const ApimapSender = require('../../src/services/apimap-sender');
+
+describe('services > apimap-sender', () => {
+  const context = new ApplicationContext();
+  const superagentRequestEndFunction = jest.fn();
+  context.init((ctx) => ctx
+    .addInstance('forestUrlGetter', jest.fn())
+    .addInstance('logger', { warn: jest.fn(), error: jest.fn() })
+    .addInstance('superagentRequest', {
+      post: () => ({ send: () => ({ set: () => ({ end: superagentRequestEndFunction }) }) }),
+    })
+    .addClass(ApimapSender));
+
+  const { apimapSender, logger } = context.inject();
+
+
+  describe('send', () => {
+    it('should post content then get response from forestUrlGetter', () => {
+      expect.assertions(1);
+      apimapSender.send();
+      expect(superagentRequestEndFunction).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('handleResult', () => {
+    describe('with no result provided', () => {
+      it('should be return early', () => {
+        expect.assertions(1);
+        expect(apimapSender.handleResult()).toBeUndefined();
+      });
+    });
+
+    describe('when having a 20X status', () => {
+      it('should be successful silently', () => {
+        expect.assertions(3);
+        [200, 202, 204].forEach((status) => {
+          expect(apimapSender.handleResult({ status })).toBeUndefined();
+        });
+      });
+
+      it('should log warning when result contains a body.warning value', () => {
+        expect.assertions(3);
+        [200, 202, 204].forEach((status) => {
+          apimapSender.handleResult({ status, body: { warning: 'Oops' } });
+          expect(logger.warn).toHaveBeenCalledWith(expect.stringMatching('Oops'));
+        });
+      });
+    });
+
+    it('should log a warning when having a 0 status', () => {
+      expect.assertions(1);
+      apimapSender.handleResult({ status: 0 });
+      expect(logger.warn).toHaveBeenCalledWith(expect.stringContaining('Are you online'));
```

Comment:
[Question/Suggestion] Maybe that's just me, but i'm not a big fan of this - I would prefer the whole logger message here.
- In a situation where this message would change, I would find weird for the test not to *always* break
- I find the test less readable this way

I guess this is opinion based, so feel free to keep it if you prefer.

Reply:
It's good advice!